src/pool_canister/src/lib.rs

```diff
@@ -43,6 +44,10 @@ pub fn init(token_id: Principal, executor: Principal) {
         cell.borrow_mut().set(executor).ok();
     });
 
+    POOL_MANAGER_PRINCIPAL.with(|cell| {
```

Comment:
There should be another principal in the init function params to be set to POOL_MANAGER_PRINCIPAL.